rust/src/builtins/scriptlet_intercept/groupadd.rs

```diff
@@ -0,0 +1,150 @@
+//! CLI handler for intercepted `groupadd`.
+
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::Result;
+use cap_std::fs::Dir;
+use clap::{Arg, Command};
+use std::io::Write;
+
+/// Entrypoint for (the rpm-ostree implementation of) `groupadd`.
+pub(crate) fn entrypoint(args: &[&str]) -> Result<()> {
+    fail::fail_point!("intercept_groupadd_ok", |_| Ok(()));
+
+    // This parses the same CLI surface as the real `groupadd`,
+    // but in the end we only extract the group name and (if
+    // present) the static GID.
+    let matches = cli_cmd().get_matches_from(args);
+    let gid: Option<u32> = match matches.value_of("gid") {
+        None => None,
+        Some(s) => Some(s.parse()?),
+    };
+    let groupname = matches
+        .value_of("groupname")
+        .expect("missing required groupname");
+
+    let rootdir = Dir::open_ambient_dir("/", cap_std::ambient_authority())?;
+    generate_sysusers_fragment(&rootdir, groupname, gid)?;
+
+    Ok(())
+}
+
+/// CLI parser, matches <https://linux.die.net/man/8/groupadd>.
+fn cli_cmd() -> Command<'static> {
+    let name = "groupadd";
+    Command::new(name)
+        .bin_name(name)
+        .about("create a new group")
+        .arg(Arg::new("force").short('f').long("force"))
+        .arg(
+            Arg::new("help")
+                .short('h')
+                .long("help")
+                .action(clap::ArgAction::Help),
+        )
+        .arg(Arg::new("gid").short('g').long("gid").takes_value(true))
+        .arg(Arg::new("key").short('K').long("key").takes_value(true))
+        .arg(Arg::new("allow_duplicates").short('o').long("non-unique"))
+        .arg(
+            Arg::new("password")
+                .short('p')
+                .long("password")
+                .takes_value(true),
+        )
+        .arg(Arg::new("system").short('r').long("system"))
+        .arg(
+            Arg::new("chroot_dir")
+                .short('R')
+                .long("root")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::new("prefix_dir")
+                .short('P')
+                .long("prefix")
+                .takes_value(true),
+        )
+        .arg(Arg::new("users").short('U').long("users").takes_value(true))
+        .arg(Arg::new("groupname").required(true))
+}
+
+/// Write a sysusers.d configuration fragment for the given group.
+///
+/// This returns whether a new fragment has been actually written
+/// to disk.
+fn generate_sysusers_fragment(rootdir: &Dir, groupname: &str, gid: Option<u32>) -> Result<bool> {
+    static SYSUSERS_DIR: &str = "usr/lib/sysusers.d";
+
+    // The filename of the configuration fragment is in fact a public
+    // API, because users may have masked it in /etc. Do not change this.
```

Comment:
Hmm.  Why would someone do that?

Reply:
This could be one approach for handling users/groups mismatches like https://github.com/coreos/fedora-coreos-tracker/issues/1201 (either as a transitory measure or a permanent override).